File: app/src/main/java/de/garten/training/depotflow/domain/StopType.java
package de.garten.training.depotflow.domain;

public enum StopType {
    PICKUP("pickup"),
    DELIVERY("delivery"),
    SERVICE("service"),
    RETURN("return");

    private final String databaseValue;

    StopType(String databaseValue) {
        this.databaseValue = databaseValue;
    }

    public String toDatabaseValue() {
        return databaseValue;
    }

    public static StopType fromDatabaseValue(String value) {
        if (value == null) {
            return SERVICE;
        }
        for (StopType kind : values()) {
            if (kind.databaseValue.equals(value)) {
                return kind;
            }
        }
        return SERVICE;
    }
}

File: app/src/main/java/de/garten/training/depotflow/domain/StopType.kt
package de.garten.training.depotflow.domain

enum class StopType(private val databaseValue: String) {
    PICKUP("pickup"),
    DELIVERY("delivery"),
    SERVICE("service"),
    RETURN("return");

    fun toDatabaseValue(): String = databaseValue

    companion object {
        @JvmStatic
        fun fromDatabaseValue(value: String?): StopType {
            return entries.firstOrNull { it.databaseValue == value } ?: SERVICE
        }
    }
}
